tlslite/messages.py

```diff
@@ -2133,3 +2127,69 @@ def parse(self, p):
 
     def write(self):
         return self.bytes
+
+
+class Heartbeat(object):
+    """
+    Handling Heartbeat messages from RFC 6520
+
+    @type message_type: int
+    @ivar message_type: type of message (response or request)
+
+    @type payload: bytearray
+    @ivar payload: payload
+
+    @type padding: bytearray
+    @ivar padding: random padding of selected length
+    """
+
+    def __init__(self):
+        self.contentType = ContentType.HEARTBEAT
+        self.message_type = 0
+        self.payload = bytearray(0)
+        self.padding = bytearray(0)
+
+    def create(self, message_type, payload, padding_length):
+        """Create heartbeat request or response with selected parameters"""
+        self.message_type = message_type
+        self.payload = payload
+        self.padding = getRandomBytes(padding_length)
+        return self
+
+    def create_response(self):
+        """Creates heartbeat response based on request."""
+        heartbeat_response = Heartbeat().create(
+            HeartbeatMessageType.HEARTBEAT_RESPONSE,
+            self.payload,
+            16)
+        return heartbeat_response
+
+    def parse(self, p):
+        """
+        Deserialize heartbeat message from parser.
+
+        We are reading only message type and payload, ignoring
+        leftover bytes (padding).
+        """
+        self.message_type = p.get(1)
+        self.payload = p.getVarBytes(2)
+        return self
+
+    def write(self):
+        """Serialise heartbeat message."""
+        w = Writer()
+        w.add(self.message_type, 1)
+        w.add(len(self.payload), 2)
+        w.bytes += self.payload
+        w.bytes += self.padding
+        return w.bytes
+
+    @property
+    def messageType(self):
```

Comment:
1. this is an internal method so it should be private
2. it's using camelCase name

Reply:
With private you mean just underscores in name? `__message_type`__message_type right?

Reply:
yes, with underscore at the begining, no, not `__message_type__` or `__message_type`, double underscore on both sides is for magic methods, double underscore at the beginning is for methods that should be hidden from child classes, so it's not necessary here